### PR TITLE
Activate lsp-pyright only when config file found

### PR DESCRIPTION
**Pull request by wyuenho**
lsp-pyright should not be activated unless a config file is located﻿

### CODE REVIEW

**Comment by seagle0128** [comment]
Why should not be activated without a config?

**Comment by wyuenho** [comment]
Because `pyright` warns when it couldn't find a config file and if the project has no config, the intention is to not use pyright. If a project hasn't declare support for it by making any configuration available, naturally there wouldn't be much code written with types. Given the high priority of this server, it will run before pretty much every other server, so there will be intended "errors" all over the place when looking at a buffer. `lsp-pyright` getting activated also causes issues with flycheck if the user has configured any `next-checkers` that will only be run if the previous checker has no warning or above errors, as is typical to flycheck checker configuration.

**Comment by seagle0128** [comment]
Basically I agree. But what about this case: users just want to use the built-in default configurations to start pyright server in some simple python scripts. If the pyright server is not started, they may report many issues because it surprises them, since the behavior is inexistent with other lsp clients.

**Comment by wyuenho** [comment]
> If the pyright server is not started, they may report many issues because it surprises them.

Document them.

>  since the behavior is inexistent with other lsp clients.

Almost all lsp clients are wrong in this regard. lsp clients should be like flycheck checkers, they should only be enabled if both the executable and configurations are found.

**Comment by seagle0128** [comment]
So could you please update the doc in this PR accordingly? And involve more people to review. Thanks!

File: README.md [hunk 1]
```diff
@@ -12,9 +12,7 @@ lsp-mode client leveraging [Pyright language server](https://github.com/microsof
 ``` emacs-lisp
 (use-package lsp-pyright
   :ensure t
-  :hook (python-mode . (lambda ()
-                          (require 'lsp-pyright)
```
Comment (seagle0128) [inline README.md:16]: If `(require 'lsp-pyright)` is removed, the client may not be registered correctly.

File: README.md [hunk 2]
```diff
@@ -34,3 +32,12 @@ lsp-mode client leveraging [Pyright language server](https://github.com/microsof
 - `python.venvPath` via `lsp-pyright-venv-path`
 
 Projects can be further configured using `pyrightconfig.json` file. For further details please see [Pyright Configuration](https://github.com/microsoft/pyright/blob/master/docs/configuration.md).
+
+### FAQ
+
+#### Why don't I see a `pyright` server running in my project when I `M-x lsp-describe-session`?
+
+`lsp-pyright` is only activated when
```
Comment (seagle0128) [inline README.md:40]: Is there any way to initialize the default config file?

Comment (wyuenho) [reply README.md:42]: No. Read the link.

**Review by seagle0128 — APPROVED** [review]

**Comment by wyuenho** [comment]
can this be merged yet?

**Comment by seagle0128** [comment]
No other reviewers approved.
And I still have concerns about the inconsistence although I've approved it. I am afraid many issues will be filed since the server isn't started automatically as before. The design of lsp is different from flycheck. Maybe the solution is to provide a command to generate the default configuration file? Is it a better idea?

**Comment by wyuenho** [comment]
Ok, I think I'm going to close this for now since it matches vscode's behavior.

**Pull request closed without merging**